kapitan/cli.py

```diff
@@ -620,6 +648,11 @@ def secret_update_validate(args, ref_controller):
                         secret_obj.update_key(awskey)
                         ref_controller[token_path] = secret_obj
 
+            elif token_path.startswith("?{vaultkv:"):
+                if not vaultkv:
+                    logger.debug("secret_update_validate: target: %s has no inventory vaultkv parameter 'auth', skipping %s", target_name, token_path)
+                    continue
+
```

Comment:
I might be mistaken, but should `secret_obj = ref_controller[token_path]` be set at this point? Same as in the cases of awskms/gkms/gpg above? Just warning if the key doesn't exist seems like it could lead to several *silent* bugs later on.